source/extensions/filters/http/grpc_http1_bridge/http1_bridge_filter.cc

```diff
@@ -18,12 +20,32 @@ namespace Extensions {
 namespace HttpFilters {
 namespace GrpcHttp1Bridge {
 
+void Http1BridgeFilter::updateContentLength(Http::RequestHeaderMap& headers, uint64_t delta) {
+  uint64_t length;
+  if (absl::SimpleAtoi(headers.getContentLengthValue(), &length)) {
+    if (length <= std::numeric_limits<uint64_t>::max() - delta) {
+      headers.setContentLength(length + delta);
+    } else {
+      ENVOY_LOG(debug, "Unabled to update content lenght for protobuf to gRPC upgrade (overflow)");
+    }
+  }
+}
+
 void Http1BridgeFilter::chargeStat(const Http::ResponseHeaderOrTrailerMap& headers) {
   context_.chargeStat(*cluster_, Grpc::Context::Protocol::Grpc, *request_stat_names_,
                       headers.GrpcStatus());
 }
 
 Http::FilterHeadersStatus Http1BridgeFilter::decodeHeaders(Http::RequestHeaderMap& headers, bool) {
+  const bool protobuf_request = Grpc::Common::isProtobufRequestHeaders(headers);
+  if (upgrade_protobuf_ && protobuf_request) {
+    do_framing_ = true;
```

Comment:
how is this value being reset? Could you test with multiple requests, one protobuf and one non protobuf and make sure framing is not applied to non protobuf request.

Reply:
As far as I understand the filter is recreated for every request, or at least that's the observable behavior with a debugger. As far as I understand there isn't anything resetting `do_bridging_` currently, so my patch builds on the assumption that something up in the chain recreates the filter (presumably FilterManager).

I'll make sure to add a test for it regardless.

Reply:
Yes filter instances are created for every stream so no reset needed.